engine/eConstr.ml

```diff
@@ -417,7 +417,7 @@ let of_binder_annot : 'a Constr.binder_annot -> 'a binder_annot =
   match Evd.MiniEConstr.unsafe_relevance_eq with
   | Refl -> fun x -> x
 
-let to_binder_annot sigma x = Context.map_annot_relevance (ERelevance.kind sigma) x
+let to_binder_annot sigma x = Context.map_annot_relevance_het (ERelevance.kind sigma) x
```

Comment:
Since it is internally homogeneous, couldn't this code use the sharing-preserving `map` by moving it at the place where we still know that `ERelevance.t` == `Sorts.relevance`? And the same for other uses of `_het`.